elements/lisk-chain/src/data_access/storage.ts

```diff
@@ -12,291 +12,537 @@
  * Removal or modification of this copyright notice is prohibited.
  */
 
+import {
+	KVStore,
+	formatInt,
+	getFirstPrefix,
+	getLastPrefix,
+	NotFoundError,
+} from '@liskhq/lisk-db';
 import { TransactionJSON } from '@liskhq/lisk-transactions';
+import { getAddressFromPublicKey } from '@liskhq/lisk-cryptography';
+
+import { AccountJSON, BlockJSON, BlockHeaderJSON } from '../types';
 
 import {
-	AccountJSON,
-	BlockJSON,
-	Storage as DBStorage,
-	TempBlock,
-} from '../types';
+	DB_KEY_BLOCKS_ID,
+	DB_KEY_BLOCKS_HEIGHT,
+	DB_KEY_TRANSACTIONS_BLOCK_ID,
+	DB_KEY_TRANSACTIONS_ID,
+	DB_KEY_TEMPBLOCKS_HEIGHT,
+	DB_KEY_ACCOUNTS_ADDRESS,
+	DB_KEY_CHAIN_STATE,
+	DB_KEY_CONSENSUS_STATE,
+} from './constants';
+import { StateStore } from '../state_store';
 
 export class Storage {
-	private readonly _storage: DBStorage;
+	private readonly _db: KVStore;
 
-	public constructor(storage: DBStorage) {
-		this._storage = storage;
+	public constructor(db: KVStore) {
+		this._db = db;
 	}
 
 	/*
 		Block headers
 	*/
-
-	public async getBlockHeaderByID(id: string): Promise<BlockJSON> {
-		const [block] = await this._storage.entities.Block.get({ id });
-
+	public async getBlockHeaderByID(id: string): Promise<BlockHeaderJSON> {
+		const block = await this._db.get<BlockHeaderJSON>(
+			`${DB_KEY_BLOCKS_ID}${id}`,
```

Comment:
Even reading here it looks better to read if we have suffix right where we use. 

`${DB_KEY_BLOCKS_ID}:${id}`